app/game/Main.hs

```diff
@@ -44,25 +46,29 @@ cliParser :: Parser CLI
 cliParser =
   subparser
     ( mconcat
-        [ command "format" (info (Format <$> parseFormat) (progDesc "Format a file"))
+        [ command "run" (info (Run <$> appOpts <**> helper) (progDesc "Run the Swarm game (default)"))
+        , command "format" (info (Format <$> parseFormat) (progDesc "Format a file"))
         , command "lsp" (info (pure LSP) (progDesc "Start the LSP"))
         , command "version" (info (pure Version) (progDesc "Get current and upstream version."))
         , command "keybindings" (info (ListKeybinding <$> initKeybindingConfig <*> printKeyMode <**> helper) (progDesc "List the keybindings"))
         ]
     )
-    <|> Run
-      <$> ( AppOpts
-              <$> seed
-              <*> scenario
-              <*> run
-              <*> autoplay
-              <*> speedFactor
-              <*> cheat
-              <*> color
-              <*> webPort
-              <*> pure gitInfo
-          )
+    <|> Run <$> appOpts
  where
+  appOpts :: Parser AppOpts
+  appOpts = do
+    userSeed <- seed
+    userScenario <- scenario
+    scriptToRun <- run
+    pausedAtStart <- paused
+    autoPlay <- autoplay
+    speed <- speedFactor
+    cheatMode <- cheat
+    colorMode <- color
+    userWebPort <- webPort
+    let repoGitInfo = gitInfo
+    return AppOpts {..}
```

Comment:
Strange, it seems GHC 9.2 and 9.4 did not understand the [`ApplicativeDo`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/applicative_do.html) I wanted to use here.

Reply:
Hmm, maybe they don't deal properly with `ApplicativeDo` + record wildcards?

Reply:
That is unfortunate.  Perhaps just put it back the way it was, but add a comment (+ an issue?) to update the code once we drop support for GHC 9.4?

Reply:
Oh! Try using `pure` instead of `return`, perhaps?

This page seems to indicate this style has been possible for a while: https://chshersh.com/recordwildcards

Reply:
The error message is wrong - the problem is in the line above containing `let`. As GHC docs note:

> _In particular, slight variations such as `return . Just $ x` or `let x = e in return x` would not be recognised._

Later versions of GHC seem to have gotten better at this.